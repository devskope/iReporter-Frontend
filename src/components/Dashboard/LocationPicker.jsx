import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import PlacesAutocomplete from 'react-places-autocomplete';

const LocationPicker = ({
  location,
  address,
  handleLocationChange,
  handleLocationSelect,
}) => (
  <Fragment>
    <PlacesAutocomplete
      value={address}
      onChange={handleLocationChange}
      onSelect={handleLocationSelect}
    >
      {({ getInputProps, suggestions, getSuggestionItemProps }) => (
        <div className="form-field form-field--location">
          <label htmlFor="edit-location">Location of Event</label>
          <input
            {...getInputProps({
              placeholder: 'Type an address',
              className: 'create-edit-form__location',
              type: 'text',
            })}
          />
          <button
            type="button"
            className="create-edit-form__location-reset btn"
          >
            Clear
          </button>
          {location && <p className="create-edit-form__geocodes">{location}</p>}
          <div className="create-edit-form__autocomplete-container">
            {suggestions.map(suggestion => (
              <div
                key={suggestion.id}
                {...getSuggestionItemProps(suggestion, {
                  className: suggestion.active
                    ? 'create-edit-form__autocomplete-item--active '
                    : 'create-edit-form__autocomplete-item',
                })}
              >
                <span>{suggestion.description}</span>
              </div>
            ))}
          </div>
        </div>
      )}
    </PlacesAutocomplete>
  </Fragment>
);

LocationPicker.defaultProps = {
  location: '',
};

LocationPicker.propTypes = {
  location: PropTypes.string,
  address: PropTypes.string.isRequired,
  handleLocationChange: PropTypes.func.isRequired,
  handleLocationSelect: PropTypes.func.isRequired,
};

export default LocationPicker;
