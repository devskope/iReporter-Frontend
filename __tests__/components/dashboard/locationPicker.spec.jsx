/* eslint-disable class-methods-use-this */
import React from 'react';
import { mount } from 'enzyme';
import LocationPicker from '../../../src/components/Dashboard/LocationPicker';

const setup = () => {
  const props = {
    location: '45.08882344, 67.96765',
    address: 'valhala',
    handleLocationChange: jest.fn(),
    handleLocationSelect: jest.fn(),
  };
  return mount(<LocationPicker {...props} />);
};

beforeAll(() => {
  global.window.google = {
    maps: {
      Marker: class {},
      Map: class {
        setTilt() {}

        fitBounds() {}
      },
      LatLngBounds: class {},
      places: {
        Autocomplete: class {},
        AutocompleteService: class {},
        PlacesServiceStatus: {
          INVALID_REQUEST: 'INVALID_REQUEST',
          NOT_FOUND: 'NOT_FOUND',
          OK: 'OK',
          OVER_QUERY_LIMIT: 'OVER_QUERY_LIMIT',
          REQUEST_DENIED: 'REQUEST_DENIED',
          UNKNOWN_ERROR: 'UNKNOWN_ERROR',
          ZERO_RESULTS: 'ZERO_RESULTS',
        },
        PlacesAutocomplete: {
          INVALID_REQUEST: 'INVALID_REQUEST',
          NOT_FOUND: 'NOT_FOUND',
          OK: 'OK',
          OVER_QUERY_LIMIT: 'OVER_QUERY_LIMIT',
          REQUEST_DENIED: 'REQUEST_DENIED',
          UNKNOWN_ERROR: 'UNKNOWN_ERROR',
          ZERO_RESULTS: 'ZERO_RESULTS',
        },
      },

      MarkerClusterer: class {},
      Geocoder: class {},
    },
  };
});

describe('LocationPicker', () => {
  it('should render without crashing', () => {
    const wrapper = setup();

    expect(wrapper.find('PlacesAutocomplete').exists()).toBe(true);
  });

  it('should match snapshot', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
