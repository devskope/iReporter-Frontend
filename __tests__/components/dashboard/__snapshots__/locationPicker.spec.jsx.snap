// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationPicker should match snapshot 1`] = `ReactWrapper {}`;
